src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp

```diff
@@ -262,63 +346,92 @@ class npc_mograine : public CreatureScript
-                    return 4 * IN_MILLISECONDS;
-                case 10:
+                    me->SetSheath(SHEATH_STATE_MELEE);
+                    events.ScheduleEvent(EVENT_MOGRAINE_EMOTE_TALK4, 800ms);
+                    break;
+                case EVENT_MOGRAINE_EMOTE_TALK4:
+                    me->AI()->Talk(SAY_MO_AB_TALK4, 200ms);
+                    events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_EMOTE_TALK, 4600ms);
+                    break;
+                case EVENT_HIGHLORD_MOGRAINE_EMOTE_TALK:
+                    summonedMograine->AI()->Talk(SAY_HM_AB_TALK1, me);
+                    events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_EMOTE1, 3400ms);
+                    break;
+                case EVENT_HIGHLORD_MOGRAINE_EMOTE1:
+                    summonedMograine->HandleEmoteCommand(EMOTE_ONESHOT_QUESTION);
+                    events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_EMOTE2, 3200ms);
+                    break;
+                case EVENT_HIGHLORD_MOGRAINE_EMOTE2:
+                    summonedMograine->HandleEmoteCommand(EMOTE_ONESHOT_POINT);
+                    events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_EMOTE3, 3200ms);
+                    break;
+                case EVENT_HIGHLORD_MOGRAINE_EMOTE3:
+                    summonedMograine->HandleEmoteCommand(EMOTE_ONESHOT_ROAR);
+                    events.ScheduleEvent(EVENT_MOGRAINE_FACING_HIGHLORD_MOGRAINE2, 3200ms);
+                    break;
+                case EVENT_MOGRAINE_FACING_HIGHLORD_MOGRAINE2:
+                    me->SetFacingToObject(summonedMograine);
+                    events.ScheduleEvent(EVENT_MOGRAINE_EMOTE_TALK5, 1200ms);
+                    break;
+                case EVENT_MOGRAINE_EMOTE_TALK5:
                     me->SetSheath(SHEATH_STATE_UNARMED);
-                    me->SetStandState(UNIT_STAND_STATE_KNEEL);
-                    Talk(5);
-                    return 2 * IN_MILLISECONDS;
-                case 11:
-                    mograine->CastSpell(me, SPELL_FORGIVENESS, false);
-                    return 1 * IN_MILLISECONDS;
-                case 12:
-                    mograine->CastSpell(me, SPELL_COSMETIC_CHAIN, true);
-                    return 0.5 * IN_MILLISECONDS;
-                case 13:
-                    mograine->AI()->Talk(2);
-                    mograine->DespawnOrUnsummon(3 * IN_MILLISECONDS);
-                    mograine->Kill(me, me, true);
-                    return 0;
+                    me->AI()->Talk(SAY_MO_AB_TALK5, 200ms);
+                    events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_CASTSPELL, 3000ms);
+                    break;
+                case EVENT_HIGHLORD_MOGRAINE_CASTSPELL:
+                    //In Blizzard's servers, after "HIGHLORD_MOGRAINE" uses this spell, "MOGRAINE" will have a visual effect of lightning hits, and the visual effect after the hit is missing here and needs to be fixed
+                    summonedMograine->CastSpell(me, SPELL_FORGIVENESS, false);
```

Comment:
```suggestion
                    summonedMograine->CastSpell(me, SPELL_FORGIVENESS);
```
By default triggered = false

Reply:
check the rest of the code too for this